src/database/database.ts

```diff
@@ -1,21 +1,61 @@
-interface Database {
+import EventModel from "./models/event-model";
+
+export default interface Database {
+  /**
+   * Create a new event. The returned `Promise` allows you to perform an action
+   * after success/failure.
+   *
+   * @param eventModel Interface containing optional fields
+   * @returns Generic JavaScript `Promise` object
+   *
+   * @example
+   * // Create a new event for a protest in LA starting on August 13th, 2020
+   * Database.createEvent({
+   *  title: "My Protest Title",
+   *  description: "My protest description.",
+   *  location: "Los Angeles, CA",
+   *  imageSource: "https://www.pngrepo.com/png/103442/180/protest.png",
+   *  dateTime: new Date(2020, 8, 13)
+   * })
+   *  .then((result) => {
+   *    // do something with result
+   *  }, (error) => {
+   *    // do something with error
+   *  });
+   */
+  createEvent(eventModel: EventModel): Promise<any>;
+
+  /**
+   * Get an existing event using its object id.
+   *
+   * @returns Generic JavaScript `Promise` object
+   *
+   * @example
+   * Database.getEvent("xWMyZ4YEGZ")
+   *  .then((result) => {
+   *    // do something with result
+   *  }, (error) => {
+   *    // do something with error
+   *  });
+   */
+  getEvent(objectId: string): Promise<any>;
```

Comment:
for the get i think it would be useful to get by zipcode, radius and tags (optional param) that way when trying to populate the list we can just take the users perferred location and call the get API with that

Reply:
We'll likely have multiple forms of `getEvents()` methods, like `getEvents(someFilter)`. That still needs to be implemented.

This one is more-so for when you want to look at the details for one specific event (like maybe when you click on one of the events in the list view).